homeassistant/components/oncue/entity.py

```diff
@@ -53,3 +53,19 @@ def _oncue_value(self) -> str:
         device: OncueDevice = self.coordinator.data[self._device_id]
         sensor: OncueSensor = device.sensors[self.entity_description.key]
         return sensor.value
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        if self._oncue_value == VALUE_UNAVAILABLE:
+            return False
+        # If the cloud is reporting that the generator is not connected
+        # this also indicates the data is not available.
+        # The battery voltage sensor reports 0.0 rather than -- hence the purpose of this check.
+        # However, the binary sensor that tracks the connection should not go unavailable.
+        if self.entity_description.key != CONNECTION_ESTABLISHED_KEY:
+            device: OncueDevice = self.coordinator.data[self._device_id]
+            conn_established: OncueSensor = device.sensors[CONNECTION_ESTABLISHED_KEY]
+            if conn_established is not None and conn_established.value == "false":
```

Comment:
I pulled the network out and the value was `--`

```
                  {'displayname': 'Network Connection Established',
                   'displayvalue': '--',
                   'groupdisplayname': 'Network Status',
                   'groupid': '52',
                   'groupname': 'NetworkStatus',
                   'id': '549405279705',
                   'name': 'NetworkConnectionEstablished',
                   'parameterdefinitionid': '872',
                   'pdiid': '4018',
                   'recordstatus': 1,
                   'value': '--',
                   'version': '221'}],
```

Reply:
Thank you, that is a bug in the change.  I’ll fix tonite and retest pulling the cable.   I did not do that after the last change.

Reply:
Have that fixed.  Ran cable pulling test and it's working as it should now.

My guess is you saw Battery Voltage not go unavailable and the Connection States binary sensor go unavailable.

Reply:
I'll give it a shot tomorrow when its light out again

Reply:
I'm having trouble getting the generator back online after pulling it.  Looks like I'm going to have to reload the firmware as that usually fixes it.  🤦

Reply:
Reloading the firmware got it back online, and there was a firmware update with a major version for it so hopefully its more reliable